pallets/funding/src/functions/misc.rs

```diff
@@ -599,22 +599,34 @@ impl<T: Config> Pallet<T> {
 		])
 	}
 
-	pub(crate) fn change_migration_status(
+	pub fn change_migration_status(
 		project_id: ProjectId,
 		user: T::AccountId,
 		status: MigrationStatus,
 	) -> DispatchResult {
+		let project_details = ProjectsDetails::<T>::get(project_id).ok_or(Error::<T>::ProjectDetailsNotFound)?;
 		let (current_status, migrations) =
-			UserMigrations::<T>::get(project_id, user.clone()).ok_or(Error::<T>::NoMigrationsFound)?;
+			UserMigrations::<T>::get((project_id, user.clone())).ok_or(Error::<T>::NoMigrationsFound)?;
+
 		let status = match status {
 			MigrationStatus::Sent(_)
 				if matches!(current_status, MigrationStatus::NotStarted | MigrationStatus::Failed) =>
 				status,
-			MigrationStatus::Confirmed if matches!(current_status, MigrationStatus::Sent(_)) => status,
+			MigrationStatus::Confirmed
+				if matches!(project_details.migration_type, Some(MigrationType::Offchain)) ||
+					(matches!(project_details.migration_type, Some(MigrationType::Pallet(_))) &&
+						matches!(current_status, MigrationStatus::Sent(_))) =>
+			{
+				UnmigratedCounter::<T>::mutate(project_id, |counter| *counter -= 1);
```

Comment:
Should we use Safe Maths here? I understand that in theory the counter should never be able to go < 0 but lets keep it safe.